### ISSUE TITLE
Remove all '-' from the yaml for spectral models

### ISSUE THREAD

**Issue by giacomov** [gammapy/gamma-astro-data-formats#3]
The current proposed YAML has a lot of extra '-' which make reading a little more cumbersome that what it could be. Let's look for example at this section:

```yaml
- PowerLaw:
    - type: SpectralModel
    - reference:
        - type: float
        - description: Value at which normalization is defined.
        - ucd: [em.wl, em.energ, em.freq]
    - index:
        - type: float
        - description: Spectral index
        - ucd: spect.index
```

To access the "powerlaw" element we need to do:

```python
with open("powerlaw.yaml") as f:

    d = yaml.load(f)

powerlaw = d[0]['PowerLaw']

```

Even more confusing, to access the only 'type' attribute of the powerlaw we need to do:
```python

print(d[0]['PowerLaw'][0]['type'])
SpectralModel
```
I believe a much cleaner way would be to remove all lists of one element (i.e., removing all "-"s) and keep only the key->value structure. For example:

```yaml

PowerLaw:
    type: SpectralModel
    reference:
        type: float
        description: Value at which normalization is defined.
        ucd: [em.wl, em.energ, em.freq]
    index:
        type: float
        description: Spectral index
        ucd: spect.index
```

This would be read as:

```python

with open("powerlaw.yaml") as f:

    d = yaml.load(f)

powerlaw = d['PowerLaw']
print(d['PowerLaw']['type'])
SpectralModel

```

and so on.

**Comment by cdeil** [comment]
@zblz @woodmd @joleroi - In case you're not watching this repo ... can you please have a look at this?

**Comment by zblz** [comment]
@giacomov - You are right, most of the minuses are unnecessary, I fixed it in eda16cb.
